src/code_80005FD0.c

```diff
@@ -1316,65 +1343,64 @@ s32 func_80008E58(s32 payerId, s32 pathIndex) {
     posZ = player->pos[2];
     stackPadding = pathIndex;
     trackSectionId = get_track_section_id(player->collision.meshIndexZX);
-    D_80163318[payerId] = trackSectionId;
+    gPlayersTrackSectionId[payerId] = trackSectionId;
     sSomeNearestWaypoint = find_closest_waypoint_track_section(posX, posY, posZ, trackSectionId, &pathIndex);
     gNearestWaypointByPlayerId[payerId] = sSomeNearestWaypoint;
     if (pathIndex) {};
     gPathIndexByPlayerId[payerId] = pathIndex;
-    func_80008DC0(pathIndex);
+    set_current_path(pathIndex);
     if (stackPadding) {};
     return pathIndex;
 }
 
-void func_80008F38(s32 playerId) {
-    f32 temp_f0;
+void update_player_completion(s32 playerId) {
+    f32 percent;
 
-    D_80164450[playerId] = (gLapCountByPlayerId[playerId] * gWaypointCountByPathIndex[0]) + sSomeNearestWaypoint;
-    temp_f0 =
-        (f32) gNearestWaypointByPlayerId[playerId] / (f32) gWaypointCountByPathIndex[gPathIndexByPlayerId[playerId]];
-    gLapCompletionPercentByPlayerId[playerId] = temp_f0;
-    gCourseCompletionPercentByPlayerId[playerId] = temp_f0;
+    gLapProgressScore[playerId] = (gLapCountByPlayerId[playerId] * gPathCountByPathIndex[0]) + sSomeNearestWaypoint;
+    percent = (f32) gNearestWaypointByPlayerId[playerId] / (f32) gPathCountByPathIndex[gPathIndexByPlayerId[playerId]];
+    gLapCompletionPercentByPlayerId[playerId] = percent;
+    gCourseCompletionPercentByPlayerId[playerId] = percent;
     gCourseCompletionPercentByPlayerId[playerId] += gLapCountByPlayerId[playerId];
 }
 
-void func_80009000(s32 playerId) {
-    s16 temp_v0;
+void yoshi_valley_cpu_path(s32 playerId) {
+    s16 previous;
 
-    temp_v0 = D_801644F8[playerId];
+    previous = gNeedToChoose[playerId];
```

Comment:
gPathChoices ? The current name is vague

Reply:
gGetPlayerPath ?

or gGetPlayerCurrentPath?

Reply:
gGetCurrentPlayerPath

Reply:
gNeedToChoose is if a playerid need to choose a path or not because it split, for example in yoshi circuit

Reply:
gNeedToChoose --> cpu_setPath? cpu_choosePath ?